### PR TITLE
PIL-975 Feature: Contacts ENS address support

### PR DESCRIPTION
**Pull request by mdjastrzebski**
Scope:
- [x] Support for input of both hex address & ENS name for address field in Contact Dialog Modal
- [x] Streamline form validation using React Hook Form + yup 
- [x] Others changes
  - [x] Fix typos
  - [x] Improve behavior of `resolveEnsName` from `utils/common`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -185,7 +187,8 @@
     "uuid": "^3.3.2",
     "vm-browserify": "0.0.4",
     "wallet-address-validator": "^0.2.4",
-    "web3": "^1.2.11"
+    "web3": "^1.2.11",
+    "yup": "^0.32.9"
```
Comment (IAmKio) [inline package.json:191]: 👍

**Review by wjkg — CHANGES_REQUESTED** [review]
I've found an edge case in the contact form that needs handling:

1. fill the name field
2. fill the address field with an ENS name that doesn't resolve
3. replace the address field with an ordinary (valid) address

result: error message is still shown and the save button doesn't work

Also, the button should probably be disabled when pressing it won't update the contact because the ENS name can't be found.

File: src/components/ContactSelector/ContactSelectorOptions.js [hunk 2]
```diff
@@ -142,9 +144,7 @@ const ContactSelectorOptions = ({
   };
 
   const handleScannerRead = (address: string) => {
-    if (isValidAddressOrEnsName(address)) {
-      selectValue({ name: address, ethAddress: address });
-    }
+    handleInputChange(address);
   };
```
[the remark this answers is not in the document]
Comment (mdjastrzebski) [reply src/components/ContactSelector/ContactSelectorOptions.js]: Good point! Fixed.

**Comment by mdjastrzebski** [comment]
Re edge case handling: nice one! Fixed.

**Comment by mdjastrzebski** [comment]
Re blocking save button: this is actually not so easy, because `errors` are populated only after validation has been triggered, which occurs only after first `onBlur` event (not from the start). So either we will duplicate validation code to handle button disabled OR we won't block it in the initial state. @wjkg wdyt?

**Comment by wjkg** [comment]
The form library docs say there should be an `isDirty` flag in the form state. Could we use that to detect the initial state?

**Comment by mdjastrzebski** [comment]
I've experimented with it, but it's still not 100% complete. E.g. after address `onBlur` event only address field will be validated, `formState.isDirty`, but name will be empty yet not validated.

Additionally from UX point of view, you might not know that name is a required field, and you might wonder why Save button is disabled. While if enable, you would get the validation message when pressing it.

IMO I would keep it as is, with active save button that can trigger helpful validation messages. @wjkg wdyt?

**Review by wjkg — APPROVED** [review]
I was thinking of consistency with e.g. new Sablier stream form, but after fixing that sticky error issue, validation on button press should be good enough. 

I've done some experimenting on my own, and `isValid` from form state _does_ take into account untouched required fields, but doesn't react to text changes without blur, so using it doesn't achieve validation while typing, anyway.

**Pull request merged**